scripts/agent-evals/src/runner/gemini-cli-runner.ts

```diff
@@ -72,21 +101,113 @@ export class GeminiCliRunner implements AgentTestRunner {
-    // through the entire history, not just the last turn
-    const found = await poll(() => {
-      if (!existsSync(this.telemetryPath)) {
+  async expectToolCalls(tools: string[]): Promise<void> {
+    await this.waitForTelemetryReady();
+
+    // We still need to poll because telemetry can take time to write each turn
+    let messages: string[] = [];
+    const success = await poll(() => {
+      messages = [];
+      let allSucceeded = true;
+      // Start at this.turnToolIndex so we only read the tools used this turn
+      const toolLogs = this.readToolLogs().slice(this.turnToolIndex);
+      const foundToolNames = toolLogs.map((log) => log.name);
+      for (const toolDef of tools) {
+        const toolName = getToolName(toolDef);
+        const matchingTool = toolLogs.find((log) => log.name === toolName);
+        if (!matchingTool) {
+          messages.push(
+            `Did not find expected tool call: "${toolName}" in the telemetry log. Found [${foundToolNames}]`,
+          );
+          allSucceeded = false;
+        } else {
+          const foundMatchingArguments = toolLogs.some(
+            (log) => log.name === toolName && toolArgumentsMatch(toolDef, log),
+          );
+          if (!foundMatchingArguments) {
+            messages.push(
+              `Tool arguments matcher "${getToolArgumentsDebug(toolDef)}" for "${toolName}" did not match any tool results in the telemetry log. All tools are: [${JSON.stringify(toolLogs)}]`,
+            );
+            allSucceeded = false;
+          }
+        }
+      }
+      return allSucceeded;
+    }, this.telemetryTimeout);
+
+    if (!success) {
+      throwFailure(messages.join("\n"));
+    }
+  }
+
+  // Implementation for this is borrowed from the Gemini CLI's test-helper
+  private async waitForTelemetryReady() {
+    // Wait for telemetry file to exist and have content
+    await poll(() => {
+      if (!fs.existsSync(this.telemetryPath)) return false;
+      try {
+        const content = readFileSync(this.telemetryPath, "utf-8");
+        // Check if file has meaningful content (at least one complete JSON object)
+        return content.includes('"event.name"');
+      } catch {
         return false;
       }
-      const content = readFileSync(this.telemetryPath, "utf-8");
-      return content.includes(eventName);
     }, this.telemetryTimeout);
+  }
+
+  // Implementation for this is borrowed from the Gemini CLI's test-helper
+  private readToolLogs(): ParsedToolLog[] {
+    const parsedLogs = this.readAndParseTelemetryLog();
+    const logs: ParsedToolLog[] = [];
 
-    if (!found) {
-      throw new Error(`Did not find expected telemetry event: "${eventName}" in the telemetry log`);
-    } else {
-      console.log(`  [FOUND] expectTelemetryEvent: ${eventName}`);
+    for (const logData of parsedLogs) {
+      // Look for tool call logs
+      if (logData.attributes && logData.attributes["event.name"] === "gemini_cli.tool_call") {
```

Comment:
```suggestion
      if (logData.attributes?.function_name && logData.attributes["event.name"] === "gemini_cli.tool_call") {
```

Saves us from needing that non null assertion below